src/config/functions.ts

```diff
@@ -38,12 +38,12 @@ export const fnSchema = {
 	macAddress: () => {
 		return macAddress.getAll(constants.macAddressPath);
 	},
-	deviceArch: async () => {
+	deviceArch: _.memoize(async () => {
```

Comment:
`_.once` makes more sense if you never plan to pass arguments:
```suggestion
	deviceArch: _.once(async () => {
```
There is the issue of if the promise fails whether you want to remember that failure and forever fail (in this case the try/catch should avoid that though). If you don't want to remember the failure then an easy solution is to use `memoizee` with `promise: true`, and the supervisor already uses memoizee elsewhere so it won't increase the size

Reply:
Oh, good catch @Page-, hadn't thought of that case. Thanks!